homeassistant/components/camera/__init__.py

```diff
@@ -702,7 +761,16 @@ async def async_internal_added_to_hass(self) -> None:
         # Avoid calling async_refresh_providers() in here because it
```

Comment:
want to add a boolean to async_refresh_providers to not write the state so it can be used here? it feels weird to duplicate code here.

Reply:
(or extract the setting logic out of the writing state when changed logic)